lib/widgets/lightbox.dart

```diff
@@ -258,9 +262,28 @@ class _ImageLightboxPageState extends State<_ImageLightboxPage> {
     // The full image is available, so display it.
     if (frame != null) return child;
 
+    // Use `BoxFit.contain` if the original image size is larger than
+    // the device screen size, otherwise `null` (default).
+    //
+    // This allows the thumbnail image to mimic the same behaviour as
+    // when the original image is loaded, which is, keep small images
+    // small (and in the center) and large images contained in the
+    // viewport.
+    BoxFit? fit;
+    final originalWidth = widget.originalWidth;
+    final originalHeight = widget.originalHeight;
+    if (originalWidth != null && originalHeight != null) {
+      final physicalSize =
+        MediaQuery.sizeOf(context) * MediaQuery.devicePixelRatioOf(context);
```

Comment:
Hmm, the meaning of `MediaQuery.sizeOf` is that it gives the size of the whole screen.

That's often the same as the size available to this widget, but it may not be — in particular we have a `SafeArea` around this, so I believe if you use a device with a notched display then this will give a larger size than what's actually available to this widget. That seems like it could lead the below logic astray.

Ideally we'd specify a single value for this option that tells `Image` to do the same thing that you explain in the comment above. Is this `BoxFit.scaleDown`?
https://main-api.flutter.dev/flutter/painting/BoxFit.html

Reply:
Ah I see, but then it's not clear there's a good way to tell `Image` about `originalWidth` and `originalHeight`.

In that case, how about we handle the sizing of the image directly? I think putting it in a SizedBox inside a FittedBox, or something along those lines, should be able to control the size it's displayed at.